File: chapter1/problem1_1.java
public class problem1_1 {
    public static void main(String[] args) {
	/*
	 * Is Unique: Implement an algorithm to determine if a string has all unique characters. What if you
	cannot use additional data structures?
	Hints: #44, #117, #132
	*/

	System.out.println(isUnique(args[0]));

	}

	public static boolean isUnique(String word){
		boolean[] characters = new boolean[256];

		for(int i = 0; i < word.length(); i++){
			if (characters[(int)word.charAt(i)]){
				return false;
			}
			characters[(int)word.charAt(i)] = true;
		}
		return true;
	}
}

File: chapter1/problem1_1.py


def IsUnique(word):
    recorded_characters = []

    for character in word:
        if character in recorded_characters: 
            return False
        recorded_characters.append(character)
    return True
    




def main():
    '''Is Unique: Implement an algorithm to determine if a string has all unique characters. What if you
    cannot use additional data structures?
    Hints: #44, #117, #132
    '''
    
    print(IsUnique("Hello World"))
    print(IsUnique("World"))





if __name__ == "__main__":
    main()



